docs/DeveloperGuide.md

```diff
@@ -329,12 +331,40 @@ This `UniqueScheduleList` would store the filtered version of the original `Addr
   * Pros: Achieved our purpose of a `ScheduleList`
   * Cons: Code duplication
 
+### Mark Student Feature
+
+This feature allows the teacher to mark a student as present for class, which increases the student's amount owed by the rates per class, while setting the student's next class date to be a week later.
+
+#### Implementation Details
+
+This command executes 3 main actions, they are:
+1. Display a cross beside the student's name in the Schedule list.
+   - `ScheduleCard.java` contains a `Label` called `markStatus` to display the cross if the student is marked.
+   - `ScheduleCard#setMarkStatus(Person person)` sets the text of `markStatus` to be `[X]` if the `person` is marked, else `[ ]`.
+
+2. Increment the money owed by the student.
+   - This action will add `ratesPerClass` field to `moneyOwed` field in `Person`.
+   - The addition of money is called through `Money#addTo(Money money)` method.
+   - To prevent integer overflow from happening, `Money#addTo(Money money)` throws a `CommandException` if it occurs.
+
+3. Set the next class to be a week later.
+   - This action will update `Class` to be `7` days later at the same `startTime` and `endTime`.
+   - Addition of days to the current `Class` date is called through `Class#addDays(int numberOfDays)` method.
+   - The next `Class` will be checked if it clashes with another `Class`. If it does not, it will be saved in `ClassStorage`. All these are called through `ClassStorage#saveClass()`.
+   - The marked `Class` will be deleted from `ClassStorage`.
+
+The following diagram illustrates how the operation works:
+
+![MarkActivityDiagram](images/DG-images/MarkActivityDiagram.png)
+
+---
+
 ### Sort-by feature
 
-This feature allows the user (teacher) to sort the students from Teacher's Pet by specified `TYPE` and `ORDER`. `ORDER` is optional
+This feature allows the user(teacher) to sort the students from Teacher's Pet by specified `TYPE` and `ORDER`. `ORDER` is optional
 and will be `ASC` when `TYPE` is `NAME` or `CLASS` and `DESC` when `TYPE` is `OWED`.
```

Comment:
This feature allows the user(teacher) to sort the students from Teacher's Pet by specified `TYPE` and `ORDER`.
`ORDER` is optional and by default, it will be set to `ASC` for `NAME` and `CLASS` sort, and `DESC` for `OWED` sort, unless otherwise specified.